### PR TITLE
Typed interfaces for composite models

### PR DESCRIPTION
**Pull request by robbr48**
Added dimensions, causality and domain specifiers for composite model interfaces.
Made sure only matching interfaces can be connected.
Disabled alignment for input/output interfaces.
Added correct alignment check for 1D bidirectional connections.
Changed colors of interface ports depending on domain.
Added overlay describing dimensions for interface ports.
Made directional connectors dashed with arrows.

### CODE REVIEW

**Comment by perost** [comment]
I'm guessing the libqjson dll-files added in the PR is a mistake?

**Comment by sjoelund** [comment]
I squashed it with the up-to-date origin. Now it can be rebased fine. Before, there were plenty of commits in the merges that already appear in the master and made reviewing the actual changes hard.

**Comment by adeas31** [comment]
Is it all good to merge now?

**Comment by adeas31** [comment]
Just saw that its still not perfect. You pushed in 16 unused perspective svg files. I remember I removed them sometime ago.

File: OMEdit/OMEditGUI/Modeling/ModelWidgetContainer.cpp [hunk 1]
```diff
@@ -4290,6 +4296,23 @@ void ModelWidget::getMetaModelConnections()
...
+        pConnectionLineAnnotation->setEndArrow(StringHandler::ArrowFilled);
+        pConnectionLineAnnotation->update();
+        pConnectionLineAnnotation->handleComponentMoved();
+    }
+    else if(pEditor->getInterfaceCausality(pConnectionLineAnnotation->getEndComponentName()) ==
+            StringHandler::getTLMCausality(StringHandler::TLMOutput)) {
+        pConnectionLineAnnotation->setLinePattern(StringHandler::LineDash);
+        pConnectionLineAnnotation->setStartArrow(StringHandler::ArrowFilled);
+        pConnectionLineAnnotation->update();
+        pConnectionLineAnnotation->handleComponentMoved();
```
Comment (adeas31) [inline OMEdit/OMEditGUI/Modeling/ModelWidgetContainer.cpp]: Are you sure you need to call,
```
pConnectionLineAnnotation->update();
pConnectionLineAnnotation->handleComponentMoved();
```

**Pull request merged**